### PR TITLE
Create CMakeList and Doxygen configuration

### PR DESCRIPTION
**Pull request by kungyao**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+html/
```
Comment (john0312) [inline .gitignore:1]: Please separate into another PR/commit.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.3)
+
+project (Group5)
+
+# add boost library
+# SET(BOOST_ROOT "./3rdparty/boost_1_76_0/")
+SET(BOOST_ROOT "D:/lib/boost_1_76_0")
```
Comment (john0312) [inline CMakeLists.txt:7]: Please do not hard code the library path.

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+# find boost library     date_time 
+FIND_PACKAGE(Boost 1.76.0 REQUIRED COMPONENTS system thread regex)
+IF(Boost_FOUND)
+  INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
+  LINK_DIRECTORIES(${Boost_LIBRARY_DIR})
+ENDIF(Boost_FOUND)
+
+# set run-time mode
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
```
Comment (john0312) [inline CMakeLists.txt:23]: Please find a more compiler neutral way to set this.

File: .gitignore (hunk 1 above)
Comment (kungyao) [reply .gitignore]: done

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+# find boost library     date_time 
+FIND_PACKAGE(Boost REQUIRED COMPONENTS system thread regex)
+IF(Boost_FOUND)
+    INCLUDE_DIRECTORIES(${Boost_INCLUDE_DIR})
+    LINK_DIRECTORIES(${Boost_LIBRARY_DIR})
+ENDIF(Boost_FOUND)
+
+# set cryptopp configure
+set(cryptopp_SRC_PATH "${PROJECT_SOURCE_DIR}/3rdparty/cryptopp")
```
[the remark this answers is not in the document]
Comment (kungyao) [reply CMakeLists.txt]: Use OpenSSL for code
